app/frontend/types.ts

```diff
@@ -1,5 +1,14 @@
 export type BIP32Path = number[]
 
+export type _Address = string & {__typeAddress: any}
```

Comment:
(probably noob) question: Do I understand correctly that `& {__typeAddress: any}` is a way to "tag" the given type leveraging type intersection so the type-checking system doesn't mix it with arbitrary strings or is there a different rationale behind this pattern? Does this pattern have a name by which I could look it up in docs/tutorials?

Reply:
Yes. At nozdormu we even have a helper type `export type Nominal<Brand, T> = T & {'__ brand': Brand}`